app/helpers/application_helper.rb

```diff
@@ -45,6 +45,11 @@ def ng_form_for(name, *args, &)
     form_for(name, *(args << options.merge(builder: AngularFormBuilder)), &)
   end
 
+  def respond_to_missing?(method_name, include_private = false)
+    (method_name.to_s.end_with?('_path',
+                                '_url') && spree.respond_to?(method_name, include_private)) || super
+  end
+
   # Pass URL helper calls on to spree where applicable so that we don't need to use
   # spree.foo_path in any view rendered from non-spree-namespaced controllers.
   def method_missing(method, *args, &)
```

Comment:
The formatting is a bit off. I think that the if-else of the method_missing block is much clearer. Anyway, this is a pretty confusing logic we apply here. It means that we can use Spree helpers in many places but in some specs it fails and you don't really know why. Your change here is good but another good change in the future would be to move everything out of `config/routes/spree.rb` into the the main application routes. They can stay in a separate file and I would move them only step by step, otherwise you break everything. But it would be a great step towards getting rid of the legacy Spree engine which we don't need anymore.